### ISSUE TITLE
Explore RPC frameworks

### ISSUE THREAD

**Issue by dsmits** [CARRIER-project/verticox#6]
Like: https://grpc.io/

**Comment by raar1** [comment]
Not sure if the small example/study I did in PerfectFit would be useful here:
<https://github.com/PerfectFit-project/virtual-coach-issues/issues/81>

**Comment by dsmits** [comment]
I looked into the following frameworks:
- [gRPC](https://grpc.io/)
- [DRPC](https://storj.github.io/drpc/): 
- [Twirp](https://github.com/verloop/twirpy)
- [Apache Thrift](https://thrift.apache.org/)

I dismissed DRPC and Thrift quite quickly. DRPC doesn't seem to have support for python, and Thrift seems to be mostly java based.

gRPC does exactly what I want to do. I don't see any big downsides except that the code doesn't look pythonic. 
Twirp looks simpler and also has python bindings. The thing is, I kind of already started implementing in gRPC. That is the main reason for now that I will go with gRPC. I will keep Twirp in the back of my mind for if it doesn't work out with gRPC.
